internal/store/db/mongo/store.go

```diff
@@ -16,37 +16,190 @@
+	client *mongo.Client
+}
+
+var mongoCollection = "store"
+
 // Store persists a stored object to the data store.
-func Store(o models.StoredObject) error {
+func (c Client) Store(o models.StoredObject) error {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+
+	_, err := c.Client.Collection(mongoCollection).InsertOne(ctx, &o)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // RetrieveFromStore gets an object from the data store.
-func RetrieveFromStore(appServiceKey string) (objects []models.StoredObject, err error) {
-	return nil, nil
+func (c Client) RetrieveFromStore(appServiceKey string) (objects []models.StoredObject, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+
+	filter := bson.D{{"appServiceKey", appServiceKey}}
+
+	// find all documents
+	cursor, err := c.Client.Collection(mongoCollection).Find(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+
+	// iterate through all documents
+	for cursor.Next(ctx) {
+		var p models.StoredObject
+		// decode the document
+		if err = cursor.Decode(&p); err != nil {
+			return nil, err
+		}
+		objects = append(objects, p)
+	}
+
+	// check if the cursor encountered any errors while iterating
+	if err = cursor.Err(); err != nil {
+		return nil, err
+	}
+
+	return objects, nil
 }
 
 // Update replaces the data currently in the store with the provided data.
-func Update(o models.StoredObject) error {
+func (c Client) Update(o models.StoredObject) error {
+	// set filters and updates
+	filter := bson.M{"id": o.ID}
+
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+
+	// update document
+	updated, err := c.Client.Collection(mongoCollection).ReplaceOne(ctx, filter, &o)
+	if err != nil {
+		return err
+	}
+
+	if updated.ModifiedCount == 0 {
+		return errors.New("no updates performed")
+	}
+
 	return nil
 }
 
 // UpdateRetryCount modifies the RetryCount variable for a given object.
-func UpdateRetryCount(id string, count int) error {
+func (c Client) UpdateRetryCount(id string, count int) error {
+	ctx, _ := context.WithTimeout(context.Background(), c.Timeout)
+
+	filter := bson.D{{"id", id}}
+	update := bson.D{
+		{"$set", bson.D{{"retryCount", count}}},
+	}
+
+	// update document
+	updated, err := c.Client.Collection(mongoCollection).UpdateOne(ctx, filter, update)
+	if err != nil {
+		return err
+	}
+
+	if updated.ModifiedCount == 0 {
+		return errors.New("no updates performed")
+	}
+
 	return nil
 }
 
 // RemoveFromStore removes an object from the data store.
-func RemoveFromStore(id string) error {
+func (c Client) RemoveFromStore(id string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+
+	deleted, err := c.Client.Collection(mongoCollection).DeleteOne(ctx, bson.M{"id": id})
+	if err != nil {
+		return err
+	}
+
+	if deleted.DeletedCount == 0 {
+		return errors.New("no deletes performed")
```

Comment:
That's not what's happening, though. If the ID wasn't found in the database, Mongo would throw its own error for that. This is when the ID is found but there was no delete performed. As I said in my other comment, it was useful for debugging my queries but since they're working and hardcoded, it really shouldn't ever happen in production.

Reply:
Agreed. Remove this check.